include/mqtt.h

```diff
@@ -87,13 +92,16 @@ SObj __cheri_compartment("MQTT")
                size_t                      incomingPublishCount,
                size_t                      outgoingPublishCount,
                const char                 *clientID,
-               size_t                      clientIDLength);
+               size_t                      clientIDLength,
+               bool                        newSession = true);
```

Comment:
This is not valid C. All public headers should compile as C and C++. We have an `__if_CXX` macro for things like this.